narwhal/consensus/src/bullshark.rs

```diff
@@ -316,13 +275,106 @@ impl Bullshark {
         Ok((Outcome::Commit, committed_sub_dags))
     }
 
+    /// Order the past leaders that we didn't already commit.
```

Comment:
I didn't really find a reason to not be under the `Bullshark` struct, as that's the only place use them and actually even inject properties of it (ex `committee` etc) so having them as standalone didn't offer much value. You think that will be an impediment? Regarding pipelined bullshark  let's discuss on the implementation because I agree that there will be overlaps